README.md

```diff
@@ -1,64 +1,88 @@
+4. Iterate over `Sting`s with the `for...of` statement.
+6. Enumerate an object's properties with the `for...in` statement.
 
-Iteration is the process of executing a set of statements **once for each element in a collection**. Prior to ES2015, there were a few ways to do this, but none were very pretty. With a `for` loop:
-```js
-var myArray = ['a', 'b', 'c', 'd', 'e', 'f', 'g'];
+## Define Iteration
 
-for (var i = 0; i < myArray.length; i++) {
-  console.log(myArray[i]);
-}
-```
+"Iteration" comes from the Latin word _iterum_ meaning "again." Iteration is
+the process of doing the same thing _again_...to each thing in a collection.
 
-With a `while` loop:
-```js
-var myArray = ['a', 'b', 'c', 'd', 'e', 'f', 'g'];
+## Discuss What's Wrong With `for` Loops; "expressivity"
+
+So what's wrong with `for` loops? They allowed us to iterate too, after all.
+The problem is that they aren't very _expressive_. You might have heard this
+term used when talking about HTML.
+
+If you saw a newspaper and in big letters at the top it said "MANKIND WALKS ON
+MOON," you, a human, would have no problem recognizing it as a header. But in
+an HTML file, a browser couldn't know the same thing.  That's one of the easons
+we "mark up" content in the HyperText Markup Language (HTML). It lets us
+document the _purpose_ of the text. With HTML we can be more _expressive_ than
+with plain text.
 
-var j = 0;
+Let's consider code where `harry_potter` performs _expelliarmus_ on each member
+of `slytherins`.
 
-while (j < myArray.length) {
-  console.log(myArray[j++]);
+```javascript
+for (let i = 0; i < slytherins.count; i = i + 1) {
+  harry_potter.expelliarmus(slytherins[i]);
 }
 ```
 
+Much like marking up content into HTML, programmers can infer that the goal is
+to iterate a collecton, but we'd like to use code that screams "HEY WE ARE
+ITERATING A COLLECTION IN ORDER HERE!." The more _expessive_, in-order way to
+travel a `String` or `Array` is a `for...of` loop.
+
 ## `for...of`
-ES2015 introduced the `for...of` statement, and, well, just compare this to the previous two snippets:
+
+ES2015 introduced the `for...of` statement. It's one and only function is to
+iterate members of `Array`s.
+
 ```js
-const myArray = ['a', 'b', 'c', 'd', 'e', 'f', 'g'];
 
-for (const element of myArray) {
-  console.log(element);
+const slytherins = ["Salazar Slytherin", "Bellatrix Black", "Draco Malfoy"];
+
+for (const snakeyWizard of slytherins) {
+  harry_potter.expelliarmus(snakeyWizard);
 }
 ```
 
-It's so much cleaner than the looping solutions — no initialization of a counter, no condition, no incrementing the counter, and no bracket notation to access elements in the array (`myArray[i]`).
+It's much cleaner than the `for` loop:
+
+* No three-part declaration divided by `;` to memoize
+* No initialization of a counter
+* No condition
+* No incrementing the counter
+* No bracket notation to access elements in the array (`myArray[i]`).
```

Comment:
I like it when we break out in-paragraph lists to actual lists like this. It's much clearer. I've seen most of us doing that, but it might not be a bad idea to formalize that as a standard.